### DISCUSSION TITLE
Fit local Resolution should revert back to orginal resolution when disconnecting

### DISCUSSION THREAD

**Discussion by nwgat** [rustdesk/rustdesk#7568] [General]
### Bug Description

Fit local Resolution does not revert back to original resolution on target when disconnecting

### How to Reproduce

Disconnect a connection when using fit resolution

### Expected Behavior

rustdesk should revert back to target's previous display resolution

### Operating system(s) on local side and remote side

Windows 11 > Windows 10

### RustDesk Version(s) on local side and remote side

1.2.4 GPU Build

### Screenshots

not needed

**Comment by rustdesk** [comment]
Thanks, reproduced on Linux->Mac

**Comment by rustdesk** [comment]
Sorry, Linux->Mac is related to our latest retina support for Mac. After turn off retina, resolution recovery works well.

We also tested -> Windows, recovery also work well.

**Comment by rustdesk** [comment]
This is marked as unreporducible, and converted to discussion. Let's keep watch if more report about this.

**Reply by uktechgeek** [reply]
I closed a previous topic, but realize now this is concerning recovering the resolution on the host machine.  This is happening without issue.  My request is mainly to remember with each connection to use the Fit Local Resolution option when connected, revert back to the original resolution with disconnect, but with each reconnect use the Fit Local Resolution option.

**Comment by uktechgeek** [comment]
_Apologies -- this was originally posted as a separate thread.  I have since deleted the other thread.  Cheers!_

I frequently use a Windows 11 laptop to connect to my Mac desktop. Scale adaptive works well, but it's not as clear on my laptop that uses 1920x1080 resolution. My Mac's monitor resolution is 2560x1440.

I found the option when connected to my Mac at the top menu bar Display Settings --> Resolution --> Fit Local Resolution (1920x1080). This option is brilliant! The remote screen's display on my laptop is crystal clear. The unfortunate problem is that when the connection is terminated and another connection is established, RustDesk does not remember the setting.

@rustdesk Please accept this request to make this a setting that is remembered with future connections. I understand the need to revert back to the default resolution when the connection is terminated, but reconnections should remember to change to Fit Local Resolution.

Cheers!

**Reply by southerndoc** [reply]
I've also experienced the same issue.  The scale adaptive doesn't come across as clear on my Windows 11 laptop.  I can't remember the resolution it's running, but it's less than my Mac Studio with Apple Studio Display resolution (which is also 2560x1440).  Haven't tried fit local resolution, but will give it a try.  If I connect with my MacBook Air to my Mac Studio, the MBA's resolution looks crystal clear even though it's a lower resolution than my ASD's resolution.  I think there is a possible issue with Windows rendering of scale adaptive.  Who knows if it can be tweaked or not.
